File: python/native_add_vehicle.py
from client import create_gateway_client
import compassiot.native.v1.native_pb2 as native


def main():
    client = create_gateway_client()

    request = native.AddVehicleRequest(
        provider_auth=native.AuthRequest(
            tesla=native.TeslaAuth(
                vin=""
            )
        )
    )

    response = client.NativeAddVehicle(request)

    print(response.message, response.add_vehicle_status)

     
if __name__ == "__main__":
    main()

File: typescript/src/native_add_vehicle.ts
import * as native from "@buf/compassiot_api.bufbuild_es/compassiot/native/v1/native_pb"

import client from "./client"

const request = new native.AddVehicleRequest({
  providerAuth: new native.AuthRequest({
    provider: {
      case: "tesla",
      value: new native.TeslaAuth({
        vin: ""
      })
    }
  })
})

const response = await client.nativeAddVehicle(request)

console.log(response.toJsonString({ prettySpaces: 2 }))
